CoreScriptsRoot/Modules/Panel3D.lua

```diff
@@ -9,36 +13,106 @@ local Panel3D = {}
 
 Panel3D.Panels = {
 	Lower = 1,
-	Hamburger = 2
+	Hamburger = 2,
+	Settings = 3
+}
+
+Panel3D.Orientation = {
+	Horizontal = 1,
+	Fixed = 2
+}
+
+Panel3D.Visibility = {
+	BelowAngleThreshold = 1,
+	Modal = 2
 }
 
 local panelDefaultVectors = {
 	[Panel3D.Panels.Lower] = CFrame.Angles(math.rad(-45), 0, 0):vectorToWorldSpace(Vector3.new(0, 0, -5)),
-	[Panel3D.Panels.Hamburger] = CFrame.Angles(math.rad(-55), 0, 0):vectorToWorldSpace(Vector3.new(0, 0, -5))
-}
-local panelLockThreshold = {
-	[Panel3D.Panels.Lower] = math.rad(-25),
-	[Panel3D.Panels.Hamburger] = math.rad(-25)
+	[Panel3D.Panels.Hamburger] = CFrame.Angles(math.rad(-55), 0, 0):vectorToWorldSpace(Vector3.new(0, 0, -5)),
+	[Panel3D.Panels.Settings] = Vector3.new(0, 0, -SETTINGS_DISTANCE)
 }
+local panelLockThreshold = math.rad(-25)
 local panelTransparencyBias = { --tuned values; raise the opacity value to this power
 	[Panel3D.Panels.Lower] = 6.5,
-	[Panel3D.Panels.Hamburger] = 8 
+	[Panel3D.Panels.Hamburger] = 8,
+	[Panel3D.Panels.Settings] = 0
 }
 local panels = {}
 
 local renderStepName = "Panel3D"
 
-
 local cursor = Instance.new("ImageLabel")
 cursor.Image = "rbxasset://textures/Cursors/Gamepad/Pointer.png"
 cursor.Size = UDim2.new(0, 8, 0, 8)
 cursor.BackgroundTransparency = 1
 cursor.ZIndex = 10
 
-local menuOpened = false
+local resetOrientationFlag = false
+local modalPanelIsOpen = false
 
+local GuiService = game:GetService("GuiService")
```

Comment:
Wouldn't it make more sense to have these two services at the top with the other two services on line 9 and 10?